### PR TITLE
JWT token from user session (behind experimental flag).

### PR DESCRIPTION
**Pull request by isoos**
- JWT token backed by a secret in Datastore entity using `HS256` algorithm
- emitted only when the experimental flag is set
- emitted only when a special header is set and response is a header value (reduces cross-origin attack surface)
- used in web_app (if enabled/available)

### CODE REVIEW

File: app/lib/account/backend.dart [hunk 1]
```diff
@@ -452,6 +465,73 @@ class AccountBackend {
     });
   }
 
+  /// Creates and signs a JWT token using HMAC-SHA256 signature.
```
Comment (jonasfj) [inline app/lib/account/backend.dart:468]: We have many kinds of tokens. If this is a CSRF token, we need to consistently use that name. Like `CsrfToken` or something.

File: app/lib/tool/neat_task/pub_dev_tasks.dart [hunk 2]
```diff
@@ -96,6 +96,13 @@ void _setupGenericPeriodicTasks() {
     task: () async => await accountBackend.deleteObsoleteSessions(),
   );
 
+  // Deletes expired tokens.
+  _daily(
+    name: 'delete-expired-tokens',
```
Comment (jonasfj) [inline app/lib/tool/neat_task/pub_dev_tasks.dart:101]: We cannot talk about generic tokens. That's too generic.

File: app/lib/account/models.dart [hunk 3]
```diff
@@ -150,6 +150,35 @@ class UserSession extends db.ExpandoModel<String> {
...
+
+  /// The expiry in the Datastore should be more into the future than in the JWT,
+  /// as deleting the entry here invalidates the JWT, regardless of the expiry
+  /// there.
+  @db.DateTimeProperty(required: true)
+  DateTime? expires;
+
+  /// The Base64-encoded secret that we use for signing the JWT signature.
+  @db.StringProperty(indexed: false)
+  String? tokenHmacSecretBase64;
```
Comment (jonasfj) [inline app/lib/account/models.dart:177]: The idea with JWT is that you wouldn't need to store state on the server.

We could use a single secret for everything.

If we store a random tokenId on the server, then I don't see why would we need to sign a JWT.

**Comment by isoos** [comment]
Closing this as we've decided to use another solution.

**Pull request closed without merging**